tests/sentry/sentry_metrics/test_cloudspanner.py

```diff
@@ -1,9 +1,32 @@
+import random
+import string
+from datetime import datetime
+from unittest.mock import patch
+
 import pytest
+from google.cloud import spanner
 
-from sentry.sentry_metrics.indexer.cloudspanner.cloudspanner import CloudSpannerIndexer, IdCodec
+from sentry.sentry_metrics.configuration import UseCaseKey
+from sentry.sentry_metrics.indexer.base import KeyResult, KeyResults
+from sentry.sentry_metrics.indexer.cloudspanner.cloudspanner import (
+    CloudSpannerIndexer,
+    IdCodec,
+    RawCloudSpannerIndexer,
+    SpannerIndexerModel,
+)
 from sentry.sentry_metrics.indexer.id_generator import get_id
 
 
+@pytest.fixture(scope="module")
+def testing_indexer():
```

Comment:
not related to this particular line, but have you tried wiring up the cloudspanner indexer with `test_all_indexers.py`? i think that would give us some more confidence that the backends behave the same in all edgecases

Reply:
That's a good idea. I will try it out.

Reply:
I will skip integrating with test_all_indexers.py for now. I tried to run it but got a bunch of failures. Since this code isn't going to be used, I am going to skip trying to debug the failures.

Reply:
I was able to run the tests from `test_all_indexers.py` locally with a clean database. One issue with these tests using spanner is that some tests write the same strings in 2 tests and then re-running the tests makes use of the same strings. I don't have the implementation of clearing the database each time the test is run. So that is a bit of pain. But functionally it works.

I am hesitant to make the change to the tests to get random strings and work with them in this PR since the PR has already gotten big.

Reply:
Nice. I think that's fine. I have the setup code here though: https://github.com/getsentry/test-factory/pull/22